users/src/main/java/io/crate/operation/user/UserManagerService.java

```diff
@@ -161,49 +159,19 @@ public void raiseMissingPrivilegeException(Privilege.Clazz clazz, @Nullable Priv
         }
     }
 
-    private static class PermissionVisitor extends AnalyzedStatementVisitor<SessionContext, Boolean> {
-
-        boolean isSuperUser(@Nullable User user) {
-            return user != null && user.roles().contains(User.Role.SUPERUSER);
-        }
-
-        private void throwUnauthorized(@Nullable User user) {
-            String userName = user != null ? user.name() : null;
-            throw new UnauthorizedException(
-                String.format(Locale.ENGLISH, "User \"%s\" is not authorized to execute statement", userName));
-        }
-
-        @Override
-        protected Boolean visitAnalyzedStatement(AnalyzedStatement analyzedStatement,
-                                                 SessionContext sessionContext) {
-            return true;
+    @Override
+    public void validateException(Throwable t, SessionContext sessionContext){
+        if (t instanceof TableUnknownException) {
+            raiseMissingPrivilegeException(Privilege.Clazz.SCHEMA, null, null, sessionContext.user());
         }
-
-        @Override
-        protected Boolean visitCreateUserStatement(CreateUserAnalyzedStatement analysis,
-                                                   SessionContext sessionContext) {
-            if (!isSuperUser(sessionContext.user())) {
-                throwUnauthorized(sessionContext.user());
-            }
-            return true;
+        if (t instanceof ResourceUnknownException) {
+            raiseMissingPrivilegeException(Privilege.Clazz.CLUSTER, null, null, sessionContext.user());
         }
-
-        @Override
-        protected Boolean visitDropUserStatement(DropUserAnalyzedStatement analysis,
-                                                 SessionContext sessionContext) {
-            if (!isSuperUser(sessionContext.user())) {
-                throwUnauthorized(sessionContext.user());
-            }
-            return true;
+        if (t instanceof ConflictException) {
```

Comment:
these exceptions happens on different sources, we must check the concrete ones, and then differ the Privilege.Cazz and ident based on that source